File: python/7 kyu/384.py
# https://www.codewars.com/kata/54fdadc8762e2e51e400032c/train/python

# Passed

def my_parse_int(strn):
    result = int(strn.strip()) if strn.strip().isdigit() else "NaN"
    return result

output = my_parse_int("1")
print(output)

File: nodejs/7 kyu/384.js
// https://www.codewars.com/kata/56971747aa359bdbf800004d/train/javascript

// Passed

function trickyDoubles(n) {
  let stringN = String(n);
  let isTrickyDouble =
    stringN.length % 2 === 0 &&
    stringN.slice(0, stringN.length / 2) === stringN.slice(stringN.length / 2);
  let result = isTrickyDouble ? n : n * 2;
  return result;
}

const output = trickyDoubles(4343);
console.log(output);